### PR TITLE
Update blueprints for `ember-mocha@0.14`

### PR DESCRIPTION
**Pull request by jbryson3**
- Upgraded the test-helper blueprint to ember-mocha 0.14 syntax
- Left the helpers blueprints in-place for backwards compatibility
  - On the fence about this one, it might be confusing to keep these unused blueprints in here without a deprecation warning
- Switched the tests to use the ember-mocha 0.14 syntax

### CODE REVIEW

**Comment by jbryson3** [comment]
I have tested these changes on a fresh ember init with ember-cli@3.1.4 - however, I'd really like to add some acceptance tests to this repo. Something akin to using `ember-cli-addon-tests` to simulate on a dummy app
- ember install ember-cli-mocha
- migrating to the new syntax
- have the dummy app run its set of tests

But so far that level of integration test has proven difficult

**Review by Turbo87 — COMMENTED** [review]
first of all thanks for working on this @jbryson3!

I think we can remove the old helpers from the `ember-cli-mocha` blueprint, but we should probably add a new `old-mocha-helpers` blueprint and copy them in there so that those that still use the older APIs have a way of generating them in the transition period.

Did you check that the new `test-helper.js` implementation using `setApplication()` also works fine when using the old APIs?

**Comment by jbryson3** [comment]
@Turbo87 I added the following sample tests in both the post 0.14 syntax, and the pre 0.14 syntax:
- Unit test
- Component Unit test
- Rendering test
- Model test
- Application test

**Review by Turbo87 — CHANGES_REQUESTED** [review]
@jbryson3 looks pretty good already

I got one more small change request though. Instead of using a `legacy-helpers` folder in the existing blueprint I think it would be better to have the helpers in the `helpers` folder of a dedicated `legacy-helpers` blueprint.

File: ember-cli-build.js [hunk 1]
```diff
@@ -9,6 +9,9 @@ module.exports = function(defaults) {
     eslint: {
       testGenerator: 'mocha',
     },
+    'ember-cli-babel': {
+      includePolyfill: true
+    }
```
[the remark this answers is not in the document]
Comment (jbryson3) [reply ember-cli-build.js]: 5 of the Old Syntax Acceptance tests were failing without it.
```
not ok 9 Chrome 67.0 - Acceptance | application - Old Syntax Acceptance | application TestLoader Failures dummy/tests/application/application-test: could not be loaded
    ---
        message: >
            regeneratorRuntime is not defined
        stack: >
            ReferenceError: regeneratorRuntime is not defined
```
A quick google search said the fix was adding the polyfill https://github.com/babel/babel-preset-env/issues/340

**Comment by jbryson3** [comment]
This PR is failing tests for reasons not related to code. Do you know any way to kick off the tests again?

image

**Comment by Turbo87** [comment]
@jbryson3 I've restarted the build :)

**Comment by topaxi** [comment]
What is the status on this? Can I help in any way? The build failure seems to be related to using node 4, which isn't used anymore on master.

**Comment by jbryson3** [comment]
Closing old pull request in a deprecated repo

**Pull request closed without merging**